daemon/README.md

```diff
@@ -0,0 +1,176 @@
+
+
+Usage
+-----
+
+You can use this container to bootstrap any Ceph daemon.
+
+* `CEPH_DAEMON` is the name the daemon you want to deploy (DEFAULT: none). Available values: `MON`, `OSD_DIRECTORY`, `OSD_CEPH_DISK`, `MDS`, `RGW`.
+* `CLUSTER` is the name of the cluster (DEFAULT: ceph)
+* `HOSTNAME` is the hostname of the machine  (DEFAULT: $(hostname))
+
+
+Deploy a monitor
+----------------
+
+Run:
+
+```
+$ sudo docker run -d --net=host \
+-v /etc/ceph:/etc/ceph \
+-v /var/lib/ceph/:/var/lib/ceph/ \
+-e CEPH_DAEMON=MON \
+-e MON_IP=192.168.0.20 \
+-e CEPH_PUBLIC_NETWORK=192.168.0.0/24 \
+ceph/daemon
+```
+
+List of available options:
+
+* `MON_IP`: IP address of the host running Docker
+* `MON_NAME`: name of the monitor (default to hostname)
+* `CEPH_PUBLIC_NETWORK`: CIDR of the host running Docker, it should be in the same network as the `MON_IP`
+* `CEPH_CLUSTER_NETWORK`: CIDR of a secondary interface of the host running Docker. Used for the OSD replication traffic
+
+
+Deploy an OSD
+-------------
+
+There are two available options:
+
+* use `OSD_CEPH_DISK` where you only specify a block device
+* use `OSD_DIRECTORY` where you specify an OSD mount point to your container
+
+
+### Ceph disk ###
+
+```
+$ sudo docker run -d --net=host \
+--privileged=true \
+-v /etc/ceph:/etc/ceph \
+-v /var/lib/ceph/:/var/lib/ceph/ \
+-v /dev/:/dev/ \
+-e CEPH_DAEMON=OSD \
+-e OSD_DEVICE=/dev/vdd \
+ceph/daemon
+```
+
+List of available options:
+
+* `OSD_DEVICE`:
+* `OSD_JOURNAL`:
+* `HOSTNAME`:
+
+If you do not want to use `--privileged=true`, please fall back on the second example.
+
+
+### Ceph OSD directory ###
+
+There are a number of environment variables which are used to configure
+the execution of the OSD:
+
+ -  `CLUSTER` is the name of the ceph cluster (defaults to `ceph`)
+
+If the OSD is not already created (key, configuration, OSD data), the
+following environment variables will control its creation:
+
+ -  `WEIGHT` is the of the OSD when it is added to the CRUSH map (default is `1.0`)
+ -  `JOURNAL` is the location of the journal (default is the `journal` file inside the OSD data directory)
+ -  `HOSTNAME` is the name of the host; it is used as a flag when adding the OSD to the CRUSH map
+
+The old option `OSD_ID` is now unused.  Instead, the script will scan for each directory in `/var/lib/ceph/osd` of the form `<cluster>-<osd-id>`.
+
+To create your OSDs simply run the following command:
+
+`docker exec <mon-container-id> ceph osd create`.
+
+
+#### Multiple OSDs ####
+
+There is a problem when attempting run run multiple OSD containers on a single docker host.  See issue #19.
+
+There are two workarounds, at present:
+* Run each OSD with a separate IP address (e.g., use the new Docker 1.5 IPv6 support)
+* Run multiple OSDs within the same container
+
+To run multiple OSDs within the same container, simply bind-mount each OSD datastore directory:
+* `docker run -v /osds/1:/var/lib/ceph/osd/ceph-1 -v /osds/2:/var/lib/ceph/osd/ceph-2`
```

Comment:
Or, since we are now providing the OSD directory option as a top-level feature, the example here should probably reflect that:  export the osd directory